### PR TITLE
Get events api integration.

### PR DESCRIPTION
**Pull request by harshjohar**
Bug Fix : Resource changed from list to object
Feature : Only enabled in `/dashboard/events`
 
Fetches 3 month data at a time
- current
- prev month
- next month

and re fetches when month changed, this way user wont face delay in a month shift and also, resource use will be minimised.

To get info about the event, can use the following interface

```ts
export interface EventResource {
    id?: number;
    branch: string;
    venue?: string;
    description?: string;
    relatedLink?: string;
}
```

### CODE REVIEW

File: components/common/EventPopup/EventPopup.tsx [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
-    subTitle: string;
-    description: string[];
+    subTitle?: string;
+    description?: string;
     imageUrl: string;
     startDate: Date;
     endDate: Date;
+    venue?: string;
     handleClose: any;
-}
```
Comment (harasees-singh) [inline components/common/EventPopup/EventPopup.tsx:17]: please pass id: number and ended: boolean as a prop as well, I need this for calling POST `/v1/events/{id}/end`

File: public/sw.js
Comment (harasees-singh) [inline public/sw.js:1]: why are you committing this file ? isn't it generated during build

File: types/eventResource.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface EventResource {
+    id?: number;
```
Comment (harasees-singh) [inline types/eventResource.ts:2]: id should not be nullable

**Review by harasees-singh — CHANGES_REQUESTED** [review]

File: public/sw.js
Comment (harshjohar) [reply public/sw.js:1]: Everytime a new package is installed, this gets generated

**Review by Ishwarendra — APPROVED** [review]
lgtm

**Review by harasees-singh — APPROVED** [review]
lgtm

**Pull request merged**